pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -450,6 +482,15 @@ private void checkIndexCreation(ColumnIndexType indexType, String column, int ca
           SegmentDirectory.Reader reader = segmentDirectory1.createReader()) {
         assertTrue(reader.hasIndexFor(column, indexType));
         assertTrue(reader.hasIndexFor(column, ColumnIndexType.FORWARD_INDEX));
+
+        // Check if the raw forward index compressionType is correct.
```

Comment:
You probably explained why this is not necessary / can't be done but I am still thinking if it should be done. 

A test to check that `index_map` file has exactly one entry for column's fwd index and the entry's start offset > what it was before reload / change

Reply:
Added validation for index_map to check if exactly one entry exists. It's not possible to check if newOffset > oldOffset because the indexmap is rewritten in alphabetical order when segmentDirectory.close() is called.